CHANGES.md

```diff
@@ -1,3 +1,67 @@
+
+# 0.11.0 (Unreleased)
+
+- [FIX] Using MongoDB query as the "gold" standard, fixed query support for `NOT` to return result sets correctly (as in MongoDB query).  Assume the following set of documents for the examples that follow:
+
+```
+{ "_id" : "fred34", "name" : "fred", "age" : 34, "pet" : "parrot" }
+{ "_id" : "fred12", "name" : "fred", "age" : 12 }
+{ "_id" : "john22", "name" : "john", "age" : 22, "pet" : "cat" }
+{ "_id" : "john44", "name" : "john", "age" : 44, "pet" : [ "hamster", "snake" ] }
+{ "_id" : "mike12", "name" : "mike", "age" : 12, "pet" : [ "cat", "dog" ] }
+{ "_id" : "mike34", "name" : "mike", "age" : 34, "pet" : [ "cat", "dog", "fish" ] }
+```
+Example One: 
+
+```
+{ "pet" : { "$not" : { "eq" : "cat" } } }
+```
+Will return the set of docments not in the result set that is returned by `{ "pet" : { "$eq" : "cat" } }`.  So if `{ "pet" : { "$eq" : "cat" } }` returns:
+
+```
+{ "_id" : "mike12", "name" : "mike", "age" : 12, "pet" : [ "cat", "dog" ] }
+{ "_id" : "mike34", "name" : "mike", "age" : 34, "pet" : [ "cat", "dog", "fish" ] }
+{ "_id" : "john22", "name" : "john", "age" : 22, "pet" : "cat" }
+```
+
+Then `{ "pet" : { "$not" : { "eq" : "cat" } } }` will return:
+
+```
+{ "_id" : "fred34", "name" : "fred", "age" : 34, "pet" : "parrot" }
+{ "_id" : "fred12", "name" : "fred", "age" : 12 }
+{ "_id" : "john44", "name" : "john", "age" : 44, "pet" : [ "hamster", "snake" ] }
+```
+
+Example Two:
+
+```
+{ "pet" : { "$not" : { "$gt": "dog" } } }
+```
+
+Will return a set of documents not in the result set that is returned by `{ "pet" : { "$gt": "dog" } }`.  So if `{ "pet" : { "$gt": "dog" } }` returns:
+
+```
+{ "_id" : "fred34", "name" : "fred", "age" : 34, "pet" : "parrot" }
+{ "_id" : "mike34", "name" : "mike", "age" : 34, "pet" : [ "cat", "dog", "fish" ] }
+{ "_id" : "john44", "name" : "john", "age" : 44, "pet" : [ "hamster", "snake" ] }
+```
+
+Then `{ "pet" : { "$not" : { "$gt": "dog" } } }` will return:
+
+```
+{ "_id" : "mike12", "name" : "mike", "age" : 12, "pet" : [ "cat", "dog" ] }
+{ "_id" : "fred12", "name" : "fred", "age" : 12 }
+{ "_id" : "john22", "name" : "john", "age" : 22, "pet" : "cat" }
+```
+
+This is different than the result set that comes from `{ "pet" : { "$lte": "dog" } }` (results below) which is the logical opposite to `{ "pet" : { "$gt": "dog" } }`.  Whereas `{ "pet" : { "$not" : { "$gt": "dog" } } }` is its "NOT-ted" opposite.
+
+```
+{ "_id" : "mike12", "name" : "mike", "age" : 12, "pet" : [ "cat", "dog" ] }
+{ "_id" : "mike34", "name" : "mike", "age" : 34, "pet" : [ "cat", "dog", "fish" ] }
+{ "_id" : "john22", "name" : "john", "age" : 22, "pet" : "cat" }
+```
+
```

Comment:
You need to succinctly describe the change here. At the moment this is more documentation. I think you should be able to do this with a single document example which before was matched and now isn't (or vice versa). Does:

> Previously, the query `{ "pet": { "$not" { "$eq": "dog" } } }` would match a document like `{ "pet" : [ "cat", "dog" ], ... }` because the array contains an object that isn't `dog`. The new behaviour is that `$not` inverts the result set, so this document would no longer be included because it has an element matching `dog`.